### ISSUE TITLE
Deployment iteration ID as environment variable

### ISSUE THREAD

**Issue by dfberry** [Azure/azure-dev#3180]
Is there an AZD ID of some kind that is used to distguish between different runs of azd that I could use to help debug the deployment? 

It seems like there is always a deployment token or guid, but what is its name and how do I get it into the env?

**Comment by rajeshkamal5050** [comment]
Don't think we have any run specific id/name/state persisted. cc @wbreza and @ellismg to clarify, if there is indeed a way to correlate.

**Comment by ellismg** [comment]
When you run `azd provision`, we create a new deployment object at subscription scope.  The name of the object is the name of your environment and a timestamp (expressed as a unix timestamp).  You can view these in the azure portal by looking at your subscription and then going to the "Deployments" view.  When `azd` prints the link to the deployment that is happening it's linking into this view.

You can also list these values with the `az` CLI using `az deployment sub list`.

Is that what you are looking for?

**Comment by dfberry** [comment]
Yes - I could use a post provision hook to add that into azd env vars?

**Comment by ellismg** [comment]
> Yes - I could use a post provision hook to add that into azd env vars? 

I think you could just add the following to your `main.bicep`:

```bicep
output DEPLOYMENT_NAME string = deployment().name
```

Since `azd` writes all of the outputs of the `main.bicep` into the .env file.

> any reason I wouldn't want to do this?

I can't think of anything that would break doing by doing this, but I guess my question would be - what's the reason for doing this?  Is there some larger goal you're after here where having a unique ID for the last deployment helps?

One other thing to note - this all depends on using Bicep as your IaC provider.  With Terraform, this wouldn't work, since `azd` just calls into `terraform` to provision the infrastructure, so you'd be stuck with whatever Terraform gives you.

**Comment by dfberry** [comment]
My 2 reasons for this information are:

1) Logging to track deployments. For complicated pipelines and hooks, which would happen in CICD, I would rather have a common tool such as Kusto to wade through logging, intead of CICD stacks such as GitHub Action output. The ID in kusto log might be able to help me track back. When I look at the container app log for startup, if the start up failed, I see that in Kusto logs. If the deploy which was meant to build the container failed, I don't see that in Kusto but instead wherever it actually happened. 

2) AZD deployments where only a service or app, out of many services/apps in the repo, is deployed. Ideally the ID could make it all the way into source code in some way that I could know which deployment item in the subscription deployment logs is associated with which app. Again - would just love to use Kusto to understand the state of deployments.

**Comment by puicchan** [comment]
As Matt mentioned, output variables set in main.bicep file are automatically stored in the .env file. It is also now documented here: https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/environment-variables-faq. 

Closing this. Please re-open if you disagree.
